File: expo-demo/modules/pull-refresh/android/src/main/java/expo/modules/pullrefresh/PullRefreshView.kt
package expo.modules.pullrefresh

import android.content.Context
import android.os.Handler
import android.os.Looper
import android.view.View
import android.view.ViewGroup
import androidx.core.widget.NestedScrollView
import androidx.recyclerview.widget.RecyclerView
import com.scwang.smart.refresh.layout.SmartRefreshLayout
import com.scwang.smart.refresh.layout.api.RefreshLayout
import com.scwang.smart.refresh.layout.constant.SpinnerStyle
import com.scwang.smart.refresh.layout.listener.OnLoadMoreListener
import com.scwang.smart.refresh.layout.listener.OnRefreshListener
import com.scwang.smart.refresh.footer.BallPulseFooter
import expo.modules.kotlin.AppContext
import expo.modules.kotlin.viewevent.EventDispatcher
import expo.modules.kotlin.views.ExpoView

/**
 * PullRefresh视图组件，集成SmartRefreshLayout功能
 * 
 * 功能对标iOS版本：
 * - 自动查找可滚动视图（ScrollView、RecyclerView等）
 * - 支持下拉刷新和上拉加载
 * - 通过Props控制状态
 * - 通过Events发送事件
 * - 线程安全
 */
class PullRefreshView(context: Context, appContext: AppContext) : ExpoView(context, appContext) {
    
    // SmartRefreshLayout实例
    private var refreshLayout: SmartRefreshLayout? = null
    
    // 刷新状态标记
    private var isRefreshing = false
    private var isLoadingMore = false
    
    // 用于追踪状态变化的key
    private var currentRefreshingKey: String = ""
    private var currentLoadingKey: String = ""
    private var currentNoMoreData: Boolean = false
    
    // 存储配置，等待找到ScrollView后应用
    private var refreshEnabled = true
    private var loadMoreEnabled = true
    private var refreshingText = "正在刷新..."
    private var loadMoreText = "加载更多..."
    private var idlePullingText = "下拉刷新"
    private var releasePullingText = "释放立即刷新"
    private var idleLoadingText = "点击或上拉加载更多"
    private var releaseLoadingText = "释放立即加载"
    private var noMoreDataText = "没有更多数据了"
    
    // 是否已经设置过刷新控件
    private var hasSetupRefreshControls = false
    
    // 主线程Handler
    private val mainHandler = Handler(Looper.getMainLooper())
    
    // 事件分发器
    private val onRefresh by EventDispatcher()
    private val onLoadMore by EventDispatcher()
    
    init {
        // 初始化时不创建SmartRefreshLayout，等待子视图添加
    }
    
    /**
     * 递归查找第一个可滚动视图（RecyclerView、ScrollView、NestedScrollView等）
     * 使用广度优先搜索算法，添加深度限制
     */
    private fun findScrollView(view: View, depth: Int = 0, maxDepth: Int = 10): View? {
        // 防止过深的递归
        if (depth >= maxDepth) return null
        
        // 检查当前视图是否是可滚动视图
        if (isScrollableView(view)) {
            return view
        }
        
        // 如果是ViewGroup，先检查直接子视图（广度优先）
        if (view is ViewGroup) {
            for (i in 0 until view.childCount) {
                val child = view.getChildAt(i)
                if (isScrollableView(child)) {
                    return child
                }
            }
            
            // 如果直接子视图中没有，再递归搜索
            for (i in 0 until view.childCount) {
                val child = view.getChildAt(i)
                findScrollView(child, depth + 1, maxDepth)?.let {
                    return it
                }
            }
        }
        
        return null
    }
    
    /**
     * 判断视图是否是可滚动视图
     */
    private fun isScrollableView(view: View): Boolean {
        return view is RecyclerView || 
               view is android.widget.ScrollView || 
               view is NestedScrollView ||
               view is android.widget.ListView ||
               view is android.widget.GridView
    }
    
    /**
     * 在找到的可滚动视图上设置SmartRefreshLayout
     */
    private fun setupRefreshControls() {
        // 防止重复设置
        if (hasSetupRefreshControls) {
            println("⚠️ PullRefresh: 刷新控件已设置，跳过重复设置")
            return
        }
        
        // 如果已经有SmartRefreshLayout，先移除
        refreshLayout?.let {
            removeView(it)
        }
        
        // 创建SmartRefreshLayout
        val layout = SmartRefreshLayout(context).apply {
            layoutParams = LayoutParams(
                LayoutParams.MATCH_PARENT,
                LayoutParams.MATCH_PARENT
            )
            
            // 使用自定义的 Lottie Header，实现类似 SmartRefreshLottie 的动画效果
            val header = LottieRefreshHeader(context)
            setRefreshHeader(header)

            // 设置上拉加载 Footer 为 SmartRefreshLayout 示例中的「球脉冲」样式
            val footer = BallPulseFooter(context).apply {
                setSpinnerStyle(SpinnerStyle.Scale)
                setAnimatingColor(android.graphics.Color.GRAY)
            }
            setRefreshFooter(footer)
            
            // 设置下拉刷新监听
            setOnRefreshListener(object : OnRefreshListener {
                override fun onRefresh(layout: RefreshLayout) {
                    this@PullRefreshView.isRefreshing = true
                    mainHandler.post {
                        this@PullRefreshView.onRefresh(emptyMap<String, Any>())
                    }
                }
            })
            
            // 设置上拉加载监听
            setOnLoadMoreListener(object : OnLoadMoreListener {
                override fun onLoadMore(layout: RefreshLayout) {
                    this@PullRefreshView.isLoadingMore = true
                    mainHandler.post {
                        this@PullRefreshView.onLoadMore(emptyMap<String, Any>())
                    }
                }
            })
            
            // 应用初始配置
            setEnableRefresh(refreshEnabled)
            setEnableLoadMore(loadMoreEnabled)
        }
        
        // 将所有现有子视图移动到SmartRefreshLayout中
        val childrenToMove = mutableListOf<View>()
        for (i in 0 until childCount) {
            childrenToMove.add(getChildAt(i))
        }
        
        removeAllViews()
        
        // 将子视图添加到SmartRefreshLayout
        childrenToMove.forEach { child ->
            layout.addView(child)
        }
        
        // 将SmartRefreshLayout添加到当前视图
        addView(layout)
        
        refreshLayout = layout
        hasSetupRefreshControls = true
        
        // 更新文字
        updateRefreshTexts()
        
        println("✅ PullRefresh: 刷新控件设置成功")
    }
    
    /**
     * 更新刷新相关文字
     */
    private fun updateRefreshTexts() {
        // 当前使用 BezierRadarHeader + BallPulseFooter，它们内部自带文案和动画效果，
        // SmartRefreshLayout 暂未提供稳定的多状态文本设置 API，因此这里暂不修改文案，
        // 如需进一步自定义，可在后续版本中通过自定义 Header/Footer 实现。
    }
    
    /**
     * 结束下拉刷新（线程安全）
     */
    fun endRefresh() {
        mainHandler.post {
            if (isRefreshing) {
                isRefreshing = false
                refreshLayout?.finishRefresh()
            }
        }
    }
    
    /**
     * 结束上拉加载（线程安全）
     */
    fun endLoadMore() {
        mainHandler.post {
            if (isLoadingMore) {
                isLoadingMore = false
                refreshLayout?.finishLoadMore()
            }
        }
    }
    
    /**
     * 没有更多数据（线程安全）
     */
    fun noMoreData() {
        mainHandler.post {
            isLoadingMore = false
            refreshLayout?.finishLoadMoreWithNoMoreData()
        }
    }
    
    /**
     * 重置没有更多数据状态（线程安全）
     */
    fun resetNoMoreData() {
        mainHandler.post {
            refreshLayout?.setNoMoreData(false)
        }
    }
    
    /**
     * 启用/禁用下拉刷新
     */
    fun setRefreshEnabled(enabled: Boolean) {
        refreshEnabled = enabled
        mainHandler.post {
            refreshLayout?.setEnableRefresh(enabled)
        }
    }
    
    /**
     * 启用/禁用上拉加载
     */
    fun setLoadMoreEnabled(enabled: Boolean) {
        loadMoreEnabled = enabled
        mainHandler.post {
            refreshLayout?.setEnableLoadMore(enabled)
        }
    }
    
    /**
     * 设置刷新文字
     */
    fun setRefreshingText(text: String) {
        refreshingText = text
        mainHandler.post {
            updateRefreshTexts()
        }
    }
    
    /**
     * 设置加载更多文字
     */
    fun setLoadMoreText(text: String) {
        loadMoreText = text
        mainHandler.post {
            updateRefreshTexts()
        }
    }
    
    /**
     * 通过key控制刷新状态
     */
    fun setRefreshingKey(key: String) {
        if (key != currentRefreshingKey && key.isNotEmpty()) {
            currentRefreshingKey = key
            endRefresh()
        }
    }
    
    /**
     * 通过key控制加载状态
     */
    fun setLoadingKey(key: String) {
        if (key != currentLoadingKey && key.isNotEmpty()) {
            currentLoadingKey = key
            endLoadMore()
        }
    }
    
    /**
     * 设置没有更多数据状态
     */
    fun setNoMoreDataFlag(flag: Boolean) {
        if (flag != currentNoMoreData) {
            currentNoMoreData = flag
            if (flag) {
                noMoreData()
            } else {
                resetNoMoreData()
            }
        }
    }
    
    /**
     * 当子视图添加时触发
     */
    override fun addView(child: View?, index: Int, params: ViewGroup.LayoutParams?) {
        if (child is SmartRefreshLayout) {
            // 如果是SmartRefreshLayout，直接添加
            super.addView(child, index, params)
        } else {
            // 如果还没有设置SmartRefreshLayout
            if (refreshLayout == null) {
                super.addView(child, index, params)
                
                // 延迟尝试设置ScrollView
                mainHandler.post {
                    trySetupScrollView()
                }
            } else {
                // 如果已经有SmartRefreshLayout，将子视图添加到它里面
                refreshLayout?.addView(child, params)
            }
        }
    }
    
    /**
     * 尝试查找并设置ScrollView
     */
    private fun trySetupScrollView() {
        // 如果已经设置过了，不需要再次设置
        if (hasSetupRefreshControls) {
            return
        }
        
        // 检查是否有子视图
        if (childCount == 0) {
            return
        }
        
        // 查找可滚动视图
        var foundScrollView = false
        for (i in 0 until childCount) {
            val child = getChildAt(i)
            if (findScrollView(child) != null) {
                foundScrollView = true
                break
            }
        }
        
        if (foundScrollView) {
            setupRefreshControls()
            println("✅ PullRefresh: 找到可滚动视图")
        } else {
            // 如果还没找到，延迟重试一次
            mainHandler.postDelayed({
                if (!hasSetupRefreshControls && childCount > 0) {
                    for (i in 0 until childCount) {
                        val child = getChildAt(i)
                        if (findScrollView(child) != null) {
                            setupRefreshControls()
                            println("✅ PullRefresh: 延迟查找成功，找到可滚动视图")
                            return@postDelayed
                        }
                    }
                    println("⚠️ PullRefresh: 未找到可滚动视图，请确保子组件包含可滚动视图（ScrollView/FlatList等）")
                }
            }, 50)
        }
    }
    
    /**
     * 清理资源
     */
    override fun onDetachedFromWindow() {
        super.onDetachedFromWindow()

        // 清理Handler消息，避免内存泄漏
        mainHandler.removeCallbacksAndMessages(null)

        // 注意：不要在这里清理 SmartRefreshLayout 的监听器
        // 原因：
        // - React Native / Expo Router 在页面切换或 Tab 切换时，View 可能会被从窗口临时移除（detach）
        // - 但 View 实例本身仍然会被复用
        // - 如果在这里把 OnRefreshListener / OnLoadMoreListener 置为 null，
        //   下次页面重新可见时 SmartRefreshLayout 仍然存在，但已经失去监听器，
        //   导致会出现「动画正常播放，但 onRefresh/onLoadMore 不再回调」的问题
        //
        // 如果后续确实需要彻底销毁时清理监听器，可以在真正销毁 View 的场景中统一处理，
        // 或者配合 onAttachedToWindow 重新绑定监听，这里先避免破坏正常回调。

        println("🔄 PullRefresh: 视图已从窗口分离，已清理 Handler 消息")
    }
}

File: expo-demo/modules/pull-refresh/ios/PullRefreshView.swift
import ExpoModulesCore
import UIKit
import MJRefresh

// PullRefresh视图组件，集成MJRefresh功能
class PullRefreshView: ExpoView {
  // 事件分发器
  let onRefresh = EventDispatcher()
  let onLoadMore = EventDispatcher()
  
  // 刷新状态标记
  private var isRefreshing = false
  private var isLoadingMore = false
  
  // 防抖时间戳（添加防护机制）
  private var lastRefreshTime: TimeInterval = 0
  private var lastLoadMoreTime: TimeInterval = 0
  private let minTriggerInterval: TimeInterval = 0.3 // 300ms最小触发间隔
  
  // 用于追踪状态变化的key
  private var currentRefreshingKey: String = ""
  private var currentLoadingKey: String = ""
  private var currentNoMoreData: Bool = false
  
  // 存储配置，等待找到ScrollView后应用
  private var refreshEnabled = true
  private var loadMoreEnabled = true
  private var refreshingText = "正在刷新..."
  private var loadMoreText = "加载更多..."
  private var idlePullingText = "下拉刷新"
  private var releasePullingText = "释放立即刷新"
  private var idleLoadingText = "点击或上拉加载更多"
  private var releaseLoadingText = "释放立即加载"
  private var noMoreDataText = "没有更多数据了"
  
  // 找到的ScrollView引用
  private weak var targetScrollView: UIScrollView?
  
  // 是否已经设置过刷新控件
  private var hasSetupRefreshControls = false
  
  // 同步队列，保证线程安全
  private let syncQueue = DispatchQueue(label: "com.pullrefresh.sync")
  
  required init(appContext: AppContext? = nil) {
    super.init(appContext: appContext)
    // 不创建自己的ScrollView，等待子视图添加
  }
  
  // 递归查找第一个UIScrollView（优化：添加深度限制）
  private func findScrollView(in view: UIView, depth: Int = 0, maxDepth: Int = 10) -> UIScrollView? {
    // 防止过深的递归
    guard depth < maxDepth else { return nil }
    
    // 优先检查当前视图
    if let scrollView = view as? UIScrollView {
      return scrollView
    }
    
    // 广度优先搜索，先检查直接子视图
    for subview in view.subviews {
      if let scrollView = subview as? UIScrollView {
        return scrollView
      }
    }
    
    // 如果直接子视图中没有，再递归搜索
    for subview in view.subviews {
      if let scrollView = findScrollView(in: subview, depth: depth + 1, maxDepth: maxDepth) {
        return scrollView
      }
    }
    
    return nil
  }
  
  // 在找到的ScrollView上设置MJRefresh
  private func setupRefreshControls(on scrollView: UIScrollView) {
    // 防止重复设置
    guard !hasSetupRefreshControls else {
      print("⚠️ PullRefresh: 刷新控件已设置，跳过重复设置")
      return
    }
    
    // 移除可能存在的旧控件
    scrollView.mj_header = nil
    scrollView.mj_footer = nil
    
    // 设置下拉刷新
    let header = MJRefreshNormalHeader { [weak self] in
      guard let self = self else { return }
      
      // 防抖检查：防止短时间内重复触发
      let currentTime = Date().timeIntervalSince1970
      if currentTime - self.lastRefreshTime < self.minTriggerInterval {
        print("⚠️ PullRefresh: 刷新触发过快，忽略本次触发（距上次 \(Int((currentTime - self.lastRefreshTime) * 1000))ms）")
        DispatchQueue.main.async {
          self.targetScrollView?.mj_header?.endRefreshing()
        }
        return
      }
      
      DispatchQueue.main.async {
        // 状态检查：如果已经在刷新中，跳过
        var shouldTrigger = false
        self.syncQueue.sync {
          if !self.isRefreshing {
            self.isRefreshing = true
            self.lastRefreshTime = currentTime
            shouldTrigger = true
          }
        }
        
        if shouldTrigger {
          print("✅ PullRefresh: 触发下拉刷新")
          self.onRefresh([:])
        } else {
          print("⚠️ PullRefresh: 已在刷新中，忽略重复触发")
          self.targetScrollView?.mj_header?.endRefreshing()
        }
      }
    }
    header.lastUpdatedTimeLabel?.isHidden = true
    header.stateLabel?.textColor = .gray
    header.stateLabel?.font = .systemFont(ofSize: 14)
    
    // 设置不同状态的文字（使用配置的文字）
    header.setTitle(idlePullingText, for: .idle)           // 默认状态
    header.setTitle(releasePullingText, for: .pulling)     // 下拉中（超过触发距离）
    header.setTitle(refreshingText, for: .refreshing)      // 正在刷新
    header.setTitle("", for: .willRefresh)                 // 即将刷新
    header.setTitle("", for: .noMoreData)                  // 没有更多数据
    
    header.isHidden = !refreshEnabled
    scrollView.mj_header = header
    
    // 设置上拉加载
    let footer = MJRefreshAutoNormalFooter { [weak self] in
      guard let self = self else { return }
      
      // 防抖检查：防止短时间内重复触发
      let currentTime = Date().timeIntervalSince1970
      if currentTime - self.lastLoadMoreTime < self.minTriggerInterval {
        print("⚠️ PullRefresh: 加载触发过快，忽略本次触发（距上次 \(Int((currentTime - self.lastLoadMoreTime) * 1000))ms）")
        DispatchQueue.main.async {
          self.targetScrollView?.mj_footer?.endRefreshing()
        }
        return
      }
      
      DispatchQueue.main.async {
        // 状态检查：如果已经在加载中，跳过
        var shouldTrigger = false
        self.syncQueue.sync {
          if !self.isLoadingMore {
            self.isLoadingMore = true
            self.lastLoadMoreTime = currentTime
            shouldTrigger = true
          }
        }
        
        if shouldTrigger {
          print("✅ PullRefresh: 触发上拉加载")
          self.onLoadMore([:])
        } else {
          print("⚠️ PullRefresh: 已在加载中，忽略重复触发")
          self.targetScrollView?.mj_footer?.endRefreshing()
        }
      }
    }
    footer.stateLabel?.textColor = .gray
    footer.stateLabel?.font = .systemFont(ofSize: 14)
    
    // 设置不同状态的文字（使用配置的文字）
    footer.setTitle(idleLoadingText, for: .idle)          // 默认状态
    footer.setTitle(releaseLoadingText, for: .pulling)    // 上拉中
    footer.setTitle(loadMoreText, for: .refreshing)       // 正在加载
    footer.setTitle("", for: .willRefresh)                // 即将加载
    footer.setTitle(noMoreDataText, for: .noMoreData)     // 没有更多数据
    
    footer.isHidden = !loadMoreEnabled
    scrollView.mj_footer = footer
    
    hasSetupRefreshControls = true
    print("✅ PullRefresh: 刷新控件设置成功")
  }
  
  // 结束刷新（线程安全）
  func endRefresh() {
    DispatchQueue.main.async { [weak self] in
      guard let self = self else { return }
      self.syncQueue.async {
        guard self.isRefreshing else { return }
        self.isRefreshing = false
      }
      self.targetScrollView?.mj_header?.endRefreshing()
    }
  }
  
  // 结束加载更多（线程安全）
  func endLoadMore() {
    DispatchQueue.main.async { [weak self] in
      guard let self = self else { return }
      self.syncQueue.async {
        guard self.isLoadingMore else { return }
        self.isLoadingMore = false
      }
      self.targetScrollView?.mj_footer?.endRefreshing()
    }
  }
  
  // 没有更多数据（线程安全）
  func noMoreData() {
    DispatchQueue.main.async { [weak self] in
      guard let self = self else { return }
      self.syncQueue.async {
        self.isLoadingMore = false
      }
      self.targetScrollView?.mj_footer?.endRefreshingWithNoMoreData()
    }
  }
  
  // 重置没有更多数据状态（线程安全）
  func resetNoMoreData() {
    DispatchQueue.main.async { [weak self] in
      guard let self = self else { return }
      self.targetScrollView?.mj_footer?.resetNoMoreData()
    }
  }
  
  // 启用/禁用下拉刷新
  func setRefreshEnabled(_ enabled: Bool) {
    refreshEnabled = enabled
    DispatchQueue.main.async { [weak self] in
      self?.targetScrollView?.mj_header?.isHidden = !enabled
    }
  }
  
  // 启用/禁用上拉加载
  func setLoadMoreEnabled(_ enabled: Bool) {
    loadMoreEnabled = enabled
    DispatchQueue.main.async { [weak self] in
      self?.targetScrollView?.mj_footer?.isHidden = !enabled
    }
  }
  
  // 设置刷新文字
  func setRefreshingText(_ text: String) {
    refreshingText = text
    DispatchQueue.main.async { [weak self] in
      if let header = self?.targetScrollView?.mj_header as? MJRefreshNormalHeader {
        // 只更新正在刷新状态的文字
        header.setTitle(text, for: .refreshing)
      }
    }
  }
  
  // 设置加载更多文字
  func setLoadMoreText(_ text: String) {
    loadMoreText = text
    DispatchQueue.main.async { [weak self] in
      if let footer = self?.targetScrollView?.mj_footer as? MJRefreshAutoNormalFooter {
        // 只更新正在加载状态的文字
        footer.setTitle(text, for: .refreshing)
      }
    }
  }
  
  // 通过key控制刷新状态
  func setRefreshingKey(_ key: String) {
    if key != currentRefreshingKey && !key.isEmpty {
      currentRefreshingKey = key
      endRefresh()
    }
  }
  
  // 通过key控制加载状态
  func setLoadingKey(_ key: String) {
    if key != currentLoadingKey && !key.isEmpty {
      currentLoadingKey = key
      endLoadMore()
    }
  }
  
  // 设置没有更多数据状态
  func setNoMoreDataFlag(_ flag: Bool) {
    if flag != currentNoMoreData {
      currentNoMoreData = flag
      if flag {
        noMoreData()
      } else {
        resetNoMoreData()
      }
    }
  }
  
  override func layoutSubviews() {
    super.layoutSubviews()
    
    // 优化：只在需要时调整子视图大小
    if bounds.size != .zero {
      for subview in subviews where subview.frame.size != bounds.size {
        subview.frame = bounds
      }
    }
  }
  
  override func insertReactSubview(_ subview: UIView!, at atIndex: Int) {
    super.insertReactSubview(subview, at: atIndex)
    addSubview(subview)
    
    // 设置子视图frame
    subview.frame = bounds
    subview.autoresizingMask = [.flexibleWidth, .flexibleHeight]
    
    // 优化：使用 DispatchQueue.main.async 而非延迟，更可靠
    DispatchQueue.main.async { [weak self] in
      self?.trySetupScrollView()
    }
  }
  
  override func removeReactSubview(_ subview: UIView!) {
    super.removeReactSubview(subview)
    
    // 修复：检查 targetScrollView 是否是被移除视图的子孙
    if let scrollView = targetScrollView {
      // 递归检查 scrollView 是否在 subview 的子树中
      if isView(scrollView, descendantOf: subview) {
        targetScrollView = nil
        hasSetupRefreshControls = false
        print("⚠️ PullRefresh: ScrollView 已被移除，清除引用")
      }
    }
    
    subview.removeFromSuperview()
  }
  
  // 辅助方法：检查一个视图是否是另一个视图的子孙
  private func isView(_ view: UIView, descendantOf parentView: UIView) -> Bool {
    var currentView: UIView? = view
    while let current = currentView {
      if current == parentView {
        return true
      }
      currentView = current.superview
    }
    return false
  }
  
  // 尝试查找并设置ScrollView（优化版本）
  private func trySetupScrollView() {
    // 如果已经找到并设置过了，检查是否还有效
    if let existingScrollView = targetScrollView {
      // 检查 ScrollView 是否还在视图树中
      if existingScrollView.superview == nil {
        print("⚠️ PullRefresh: 检测到 ScrollView 已从视图树移除")
        targetScrollView = nil
        hasSetupRefreshControls = false
      } else {
        // ScrollView 仍然有效，不需要重新设置
        return
      }
    }
    
    // 在子视图中查找ScrollView
    for subview in subviews {
      if let scrollView = findScrollView(in: subview) {
        targetScrollView = scrollView
        setupRefreshControls(on: scrollView)
        print("✅ PullRefresh: 找到 ScrollView 类型: \(type(of: scrollView))")
        return
      }
    }
    
    // 如果还是没找到，延迟重试一次（React Native 视图树可能还在构建中）
    if targetScrollView == nil {
      DispatchQueue.main.asyncAfter(deadline: .now() + 0.05) { [weak self] in
        guard let self = self, self.targetScrollView == nil else { return }
        
        for subview in self.subviews {
          if let scrollView = self.findScrollView(in: subview) {
            self.targetScrollView = scrollView
            self.setupRefreshControls(on: scrollView)
            print("✅ PullRefresh: 延迟查找成功，找到 ScrollView")
            return
          }
        }
        
        print("⚠️ PullRefresh: 未找到 ScrollView，请确保子组件包含可滚动视图（ScrollView/FlatList等）")
      }
    }
  }
  
  // 当视图添加到窗口时，再次尝试查找ScrollView
  override func didMoveToWindow() {
    super.didMoveToWindow()
    if window != nil {
      trySetupScrollView()
    }
  }
  
  // 清理资源
  deinit {
    // 移除 MJRefresh 控件
    targetScrollView?.mj_header = nil
    targetScrollView?.mj_footer = nil
    print("🔄 PullRefresh: 视图已销毁，清理资源")
  }
}
